warp10/src/main/java/io/warp10/script/functions/MAP.java

```diff
@@ -69,8 +69,18 @@ public Object apply(WarpScriptStack stack) throws WarpScriptException {
 
     List<Object> params = (List<Object>) top;
 
+    if ( 2 > params.size()) {
+      throw new WarpScriptException(getName() + " needs a list of at least 2 parameters as input.");
+    }
+
+    if (!(params.get(params.size() - 1) instanceof Number)) {
```

Comment:
You are assuming that params.size() is 2, which might not be the case. How about adding 0s until params.size() is 5 if params.size() is initially less than 5?

Reply:
It would not work with the original signature of MAP that is still supported (there can be any number of GTS or lists before the mapper), unless we compare with (nseries + 3) instead of 5.

Reply:
In that case check the last 1-3 parameters and complete with 0s for the missing ones.